### ISSUE TITLE
EventSubMiddleware not working with Express Router

### ISSUE THREAD

**Issue by AshuGuptaGamer** [twurple/twurple#514]
## Bug Report

`EventSubMiddleware` not working with Express `Router`.
The middleware is registering the subscriptions but the `POST` route it was supposed to listen on is returning 404 not found.

### Code

file: `registerTwitchRoute.js`
```js
  const app = Router();
  await api.eventSub.deleteAllSubscriptions();
  const middleware = new EventSubMiddleware({
    apiClient: api,
    hostName: config.twitch.hostUrl, // 'example.com'
    pathPrefix: config.twitch.rootRoute, // '/twitch'
    secret: secret,
    legacySecrets: false,
    usePathPrefixInHandlers: false
  });
  middleware.apply(app);
  return { app, middleware };
```
file: `express.js`
```js
  const { app, middleware } = registerTwitchRoute();
  app.use('/twitch', app);

  app.login(3000, () => {
    middleware.markAsReady();
    console.log(`Listening on port 3000...`);
  })
```

### Expected behavior

Should work on Express Router implementations.

### Actual Behavior

Doesn't accept post requests when using Express Router implementations.

### Environment

- Version: 6.2.1
- Node version: 16.13.0
- Operating system: Windows 11 OS Build 22621.1848

###Images

![image](https://github.com/twurple/twurple/assets/59139918/a7cbc8ee-7dbe-43d8-8c37-4cb67ceb28c4)
![image](https://github.com/twurple/twurple/assets/59139918/c99540b7-15b1-4985-8489-7b45eb6f11bf)

**Comment by d-fischer** [comment]
I've noticed two issues with your code:

- You named both your app and your router `app`, which means you probably have problems accessing the app properly.
- You called `app.login()` where you probably meant to call `app.listen()`.

After fixing these two issues (and adapting to use my own test server), I was able to successfully use your router setup to receive EventSub notifications.

Maybe your `rootRoute` might not be what you think it is? Please supply a full reproduction case I don't have to adapt at all.

**Comment by Sylphias** [comment]
Actually I am facing the same issue as well. Even though I do have it set up after `app.listen` and have verified that the routes have been registered on the express router.
![image](https://github.com/twurple/twurple/assets/5444382/2d109fe0-c168-4c9c-b9a2-d8d4e931c2f6)
but I'm still getting a 404 on redemption events callbacks. I used to work with the reverse proxy edition and that worked perfectly fine.
![image](https://github.com/twurple/twurple/assets/5444382/dc260116-7997-438e-b442-46b2a22cf02e)

Here is the section I code I use to setup the middleware (I've tried it with strictHostCheck true and false. so that isnt the problem, envs are just placeholders but I have backup code to prevent undefined.)

```
EventSubMiddleware({
      apiClient,
      hostName: process.env.TUNNEL_HOST_NAME || "",
      secret: process.env.EVENT_SECRET || "",
      strictHostCheck: false,
      pathPrefix: "/twitch",
      logger: {
        minLevel: 'debug'
      }
    });
 ```
